execution/relation.go

```diff
@@ -168,13 +174,74 @@ func (rel *Like) Apply(ctx context.Context, variables octosql.Variables, left, r
 			leftValue.Show(), rightValue.Show(), leftValue.GetType(), rightValue.GetType())
 	}
 
-	match, err := regexp.MatchString(rightValue.AsString(), leftValue.AsString())
+	patternString, err := likePatternToRegexp(rightValue.AsString())
+	if err != nil {
+		return false, errors.Wrapf(err, "couldn't transform LIKE pattern %v to regexp", patternString)
+	}
+
+	match, err := regexp.MatchString(patternString, leftValue.AsString())
 	if err != nil {
 		return false, errors.Wrapf(err, "couldn't match string in like relation with pattern %v", rightValue)
 	}
 	return match, nil
 }
 
+//we assume that the escape character is '\'
+func likePatternToRegexp(pattern string) (string, error) {
+	length := len(pattern)
+	if pattern[length-1] == LikeEscape {
+		return "", errors.New("the pattern cannot end with an escape character")
+	}
+
+	var sb strings.Builder
+
+	sb.WriteRune('^') // match start
+
+	for i, r := range pattern {
```

Comment:
In my opinion you can just store a boolean "escaping", which gets set to true whenever you encounter a functioning escape character.

Now whenever you enter the loop and escaping=true, then just rewrite the current character literally.

If you exit the loop with escaping=true, error, because you have a functioning escape character as the last codepoint in the string.